controllers/cluster.weave.works/existinginframachine_controller.go

```diff
@@ -589,6 +608,72 @@ func (a *ExistingInfraMachineReconciler) update(ctx context.Context, c *existing
 	return nil
 }
 
+func (a *ExistingInfraMachineReconciler) renewKubeadmCerts(ctx context.Context, installer *os.OS) error {
+	var certificateKey string
+
+	// Renewal of kubeadm-certs is done in two plans so that the certificate key, printed
+	// in the output of `kubeadm alpha certs certificate-key` can be passed in the upload certs plan
+	getCertKeyPlan, err := recipe.BuildGetKubeadmCertKeyPlan(ctx, &certificateKey)
+	if err != nil {
+		log.Errorf("failed to build get kubeadm cert key plan, err: %s", err)
+		return err
+	}
+	// The certificateKey is populated after applying the get kubeadm cert key plan
+	// Trim the newline suffix that comes with the output of the cli command
+	_, err = getCertKeyPlan.Apply(ctx, installer.Runner, plan.EmptyDiff())
+	if err != nil {
+		log.Errorf("Apply of Plan failed:\n%s\n", err)
+		return err
+	}
+	certificateKey = strings.TrimSuffix(certificateKey, "\n")
+	log.Infof(fmt.Sprintf("uploading kubeadm certs with cert key: %s", certificateKey))
```

Comment:
Is this a public key? We should avoid adding sensitive keys to logs.

Reply:
Good point, was useful for debugging, but also it is printed in the output of the `kubeadm` command when the plan is applied. Will need to check if that's conceilable.

Reply:
Even then I realise the output is needed so that we know what cert key is used, and pass it to `kubeadm init phase upload-certs --upload-certs --certificate-key` and to update the `wks-controller-secrets` with it..
It is possible to avoid that if it could be generated on the controller.

Reply:
Refactored it to generate the cert key as in https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/alpha/certs.go#L189